### PR TITLE
[AWS] [Lambda] Spec

### PR DESCRIPTION
**Pull request by nagypeterjob**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Feature

**What is the current behavior? (You can also link to an open issue here)**

Missing spec

**What is the new behavior (if this is a feature change)?**

Added lambda autocompletion

**Additional info:**

* [x] add-layer-version-permission
* [x] add-permission
* [x] create-alias
* [x] create-code-signing-config
* [x] create-event-source-mapping
* [x] create-function
* [x] delete-alias
* [x] delete-code-signing-config
* [x] delete-event-source-mapping
* [x] delete-function
* [x] delete-function-code-signing-config
* [x] delete-function-concurrency
* [x] delete-function-event-invoke-config
* [x] delete-layer-version
* [x] delete-provisioned-concurrency-config
* [x] get-account-settings
* [x] get-alias
* [x] get-code-signing-config
* [x] get-event-source-mapping
* [x] get-function
* [x] get-function-code-signing-config
* [x] get-function-concurrency
* [x] get-function-configuration
* [x] get-function-event-invoke-config
* [x] get-layer-version
* [x] get-layer-version-by-arn
* [x] get-layer-version-policy
* [x] get-policy
* [x] get-provisioned-concurrency-config
* [x] invoke
* [x] list-aliases
* [x] list-code-signing-configs
* [x] list-event-source-mappings
* [x] list-function-event-invoke-configs
* [x] list-functions
* [x] list-functions-by-code-signing-config
* [x] list-layer-versions
* [x] list-layers
* [x] list-provisioned-concurrency-configs
* [x] list-tags
* [x] list-versions-by-function
* [x] publish-layer-version
* [x] publish-version
* [x] put-function-code-signing-config
* [x] put-function-concurrency
* [x] put-function-event-invoke-config
* [x] put-provisioned-concurrency-config
* [x] remove-layer-version-permission
* [x] remove-permission
* [x] tag-resource
* [x] untag-resource
* [x] update-alias
* [x] update-code-signing-config
* [x] update-event-source-mapping
* [x] update-function-code
* [x] update-function-configuration
* [x] update-function-event-invoke-config
* [x] wait

### CODE REVIEW

**Comment by QuiiBz** [comment]
Hello, could you update this branch by merging master in it? We updated how the completion specs are compiled, and so the `specs` directory doesn't exist anymore.

**Comment by nagypeterjob** [comment]
> Hello, could you update this branch by merging master in it? We updated how the completion specs are compiled, and so the `specs` directory doesn't exist anymore.

Well, I did merge master into the branch, but I guess it wasn't a great idea after all. This PR was opened against `mschrange/aws` which is also outdated compared to `master`. I think we should open this PR against `master`

**Comment by QuiiBz** [comment]
> 
> Well, I did merge master into the branch, but I guess it wasn't a great idea after all. This PR was opened against `mschrange/aws` which is also outdated compared to `master`. I think we should open this PR against `master`

Sorry for the delay. I agree with you, could you re-open this PR against the current `master`?

**Comment by QuiiBz** [comment]
I merged master into this PR and fixed the conflicts

File: specs/aws/iot-data.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
+var completionSpec = {
```
Comment (QuiiBz) [inline specs/aws/iot-data.js:1]: Any reason why there is this file in the lambda PR?

File: dev/git.ts [hunk 2]
```diff
@@ -2146,10 +2146,10 @@ export const completionSpec: Fig.Spec = {
         },
       ],
       args: {
-        name: "commit",
+        name: "file|commit",
         isOptional: true,
         suggestions: [{ name: "HEAD" }],
-        generators: gitGenerators.commits,
+        generators: [gitGenerators.files_for_staging, gitGenerators.commits],
```
Comment (QuiiBz) [inline dev/git.ts:2152]: Let's remove this from the PR since it's not related to the lambda spec.

**Review by QuiiBz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**